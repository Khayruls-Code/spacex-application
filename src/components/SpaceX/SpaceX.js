import React, { useState } from 'react';
import SearchFilter from '../SearchFilter/SearchFilter';

const SpaceX = () => {

  const [showSpaceData, setShowSpaceData] = useState([])

  return (
    <div className='container mx-auto px-4'>
      <SearchFilter setShowSpaceData={setShowSpaceData} />
      <div className="grid sm:grid-cols-2 lg:grid-cols-3 gap-6">
        {
          showSpaceData?.map(singleData => <div key={singleData?.rocket?.rocket_id + (Math.random() * 10000)}
            className='bg-gradient-to-r from-transparent_violet to-transparent_purple rounded-md'
          >
            <img className={singleData?.links?.flickr_images.length ? "w-full h-60 object-cover rounded-t-md" : 'w-full h-60 object-contain py-3 bg-white rounded-t-md'} src={singleData?.links?.flickr_images[0] || singleData?.links.mission_patch} alt="" />
            <div className='p-4 relative'>
              <p className={singleData?.launch_success ? "bg-green absolute -top-3 right-0 px-3 text-md font-medium text-white rounded-sm w-fit shadow-lg" : "bg-red absolute -top-3 right-0 px-3 text-md font-medium text-white rounded-sm w-fit shadow-lg"}>{singleData?.launch_success ? "Success" : 'Failure'}</p>
              <h1 className='text-3xl font-semibold font-mono text-white'>{singleData?.rocket?.rocket_name}</h1>
              <p className='text-white my-2 tracking-wide'>{singleData?.details?.slice(0, 100)}</p>
              <div className='flex items-center gap-4'>
                <h2 className='text-md text-white'><span className='font-bold'>Launch Date:</span> {singleData?.launch_date_local?.split('T')[0]}</h2>
                <h3 className='text-black bg-white px-3 rounded-full font-semibold'>{singleData?.upcoming && "Upcoming"}</h3>
              </div>
              <div>

              </div>
            </div>
          </div>)
        }
      </div>
      {
        !showSpaceData.length && <h1 className='text-white text-4xl  flex items-center justify-center mt-32'>Data Not Found</h1>
      }
    </div >
  );
};

export default SpaceX;