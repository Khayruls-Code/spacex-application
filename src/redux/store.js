import { configureStore } from '@reduxjs/toolkit'
import spaceXDataSlice from './features/spaceXDataSlice'

const store = configureStore({
  reducer: {
    spaceXData: spaceXDataSlice,
  },
})

export default store;