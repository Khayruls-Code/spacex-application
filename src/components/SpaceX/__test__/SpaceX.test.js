import React from 'react'
import ReactDOM from 'react-dom'
import SpaceX from '../SpaceX'
import { Provider } from 'react-redux'
import store from '../../../redux/store'
import renderer from 'react-test-renderer'

it('SpaceX renders without crashing',
() => {
  const div = document.createElement('div')
  ReactDOM.render(
    <Provider store={store}>
      <SpaceX />
    </Provider>,
div)
})

it("matchs SpaceX snapshot",
() => {
  const tree = renderer.create(
    <Provider store={store}>
      <SpaceX />
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})