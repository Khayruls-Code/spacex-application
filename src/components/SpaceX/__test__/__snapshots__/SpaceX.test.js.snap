// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs SpaceX snapshot 1`] = `
<div
  className="container mx-auto px-4"
>
  <div
    className="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 mb-4 gap-y-4 gap-x-4 md:gap-x-8"
    data-testid="searchFilterId"
  >
    <input
      className="py-1.5 px-2 outline-none rounded-sm font-semibold w-full"
      onChange={[Function]}
      placeholder="Search By Rocket Name..."
      type="text"
    />
    <select
      className="py-1.5 px-2 outline-none rounded-sm font-semibold w-full"
      onChange={[Function]}
    >
      <option
        disabled="disabled"
        selected="true"
      >
        Filter By Launch Date
      </option>
      <option
        value="Last Week"
      >
        Last Week
      </option>
      <option
        value="Last Month"
      >
        Last Month
      </option>
      <option
        value="Last Year"
      >
        Last Year
      </option>
    </select>
    <select
      className="py-1.5 px-2 outline-none rounded-sm font-semibold w-full"
      onChange={[Function]}
    >
      <option
        disabled="disabled"
        selected="true"
      >
        Filter By Launch Status
      </option>
      <option
        value="Success"
      >
        Success
      </option>
      <option
        value="Failure"
      >
        Failure
      </option>
    </select>
    <select
      className="py-1.5 px-2 outline-none rounded-sm font-semibold w-full"
      onChange={[Function]}
    >
      <option
        disabled="disabled"
        selected="true"
      >
        Upcoming Or Retrospective
      </option>
      <option
        value="Upcoming"
      >
        Upcoming
      </option>
      <option
        value="Retrospective"
      >
        Retrospective
      </option>
    </select>
  </div>
  <div
    className="grid sm:grid-cols-2 lg:grid-cols-3 gap-6"
  />
  <h1
    className="text-white text-4xl  flex items-center justify-center mt-32"
  >
    Data Not Found
  </h1>
</div>
`;

exports[`matchs snapshot 1`] = `
<div
  className="container mx-auto px-4"
>
  <div
    className="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 mb-4 gap-y-4 gap-x-4 md:gap-x-8"
    data-testid="searchFilterId"
  >
    <input
      className="py-1.5 px-2 outline-none rounded-sm font-semibold w-full"
      onChange={[Function]}
      placeholder="Search By Rocket Name..."
      type="text"
    />
    <select
      className="py-1.5 px-2 outline-none rounded-sm font-semibold w-full"
      onChange={[Function]}
    >
      <option
        disabled="disabled"
        selected="true"
      >
        Filter By Launch Date
      </option>
      <option
        value="Last Week"
      >
        Last Week
      </option>
      <option
        value="Last Month"
      >
        Last Month
      </option>
      <option
        value="Last Year"
      >
        Last Year
      </option>
    </select>
    <select
      className="py-1.5 px-2 outline-none rounded-sm font-semibold w-full"
      onChange={[Function]}
    >
      <option
        disabled="disabled"
        selected="true"
      >
        Filter By Launch Status
      </option>
      <option
        value="Success"
      >
        Success
      </option>
      <option
        value="Failure"
      >
        Failure
      </option>
    </select>
    <select
      className="py-1.5 px-2 outline-none rounded-sm font-semibold w-full"
      onChange={[Function]}
    >
      <option
        disabled="disabled"
        selected="true"
      >
        Upcoming Or Retrospective
      </option>
      <option
        value="Upcoming"
      >
        Upcoming
      </option>
      <option
        value="Retrospective"
      >
        Retrospective
      </option>
    </select>
  </div>
  <div
    className="grid sm:grid-cols-2 lg:grid-cols-3 gap-6"
  />
  <h1
    className="text-white text-4xl  flex items-center justify-center mt-32"
  >
    Data Not Found
  </h1>
</div>
`;
